exporters/secret/exporter.go

```diff
@@ -102,14 +111,69 @@ func (e *Exporter) calculateExpiry(ch chan<- prometheus.Metric, secret v1.Secret
 
 		for _, cert := range certs {
 			timestamp := float64(cert.NotAfter.Unix())
-			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, certType)
+			issuer := cert.Issuer.String()
+			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, issuer, certCRName)
 		}
 	}
 	e.logger.Log("info", fmt.Sprintf("added secret %s/%s to the metrics", secretNamespace, secretName))
 
 	return nil
 }
 
+func (e *Exporter) certificateGroupVersionResources() ([]schema.GroupVersionResource, error) {
+	groupVersions := []schema.GroupVersionResource{}
+
+	_, resources, err := e.k8sClient.ServerGroupsAndResources()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	for _, rl := range resources {
+		for _, r := range rl.APIResources {
+			if r.Name == "certificates" && strings.HasPrefix(rl.GroupVersion, "cert-manager.io/") {
+				s := schema.FromAPIVersionAndKind(rl.GroupVersion, r.Name)
+				groupVersions = append(groupVersions, schema.GroupVersionResource{
+					Group:    s.Group,
+					Version:  s.Version,
+					Resource: r.Name,
+				})
+			}
+		}
+	}
+
+	return groupVersions, nil
+}
+
+func (e *Exporter) findCertCRName(secretName, secretNamespace string) (string, error) {
+	certificateGroupVersionResources, err := e.certificateGroupVersionResources()
+	if err != nil {
+		e.logger.Log("error", microerror.Mask(err))
+		return "", err
+	}
+	if len(certificateGroupVersionResources) == 0 {
+		e.logger.Log("info", "cert-manager Certificate not available, skipping secret collection")
+		return "", nil
+	}
+
+	for _, gvr := range certificateGroupVersionResources {
+		certs, err := e.dynamicClient.Resource(gvr).Namespace(secretNamespace).List(e.ctx, metav1.ListOptions{})
+		if err != nil {
+			e.logger.Log("error", microerror.Mask(err))
+		}
+		for _, cert := range certs.Items {
+			name, found, err := unstructured.NestedString(cert.UnstructuredContent(), "spec", "secretName")
+			if err != nil {
+				e.logger.Log("error", microerror.Mask(err))
+			}
+			if found && name == secretName {
+				return cert.GetName(), nil
```

Comment:
Do we need to care about the namespace here? What if two certs exist with the same name in different namespaces? Is this possible?

Reply:
Just checked cert-manager docs, looks like the `Secret` of a `Certificate` is always created in the same namespace as the `Certficate`.

It is possible to have `Certificates` with the same name in different namespaces, but I doubt they're interfering in this case. In Line 159 the code only lists `Certificates` in the namespace of the `Secret`.